src/audio/template_comp/template_comp.c

```diff
@@ -0,0 +1,311 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <rtos/init.h>
+#include "template_comp.h"
+
+LOG_MODULE_REGISTER(template_comp, CONFIG_SOF_LOG_LEVEL);
+
+/* Use e.g. command uuidgen from package uuid-runtime, add it to
+ * uuid-registry.txt in SOF top level.
+ */
+SOF_DEFINE_REG_UUID(template_comp);
+
+DECLARE_TR_CTX(template_comp_tr, SOF_UUID(template_comp_uuid), LOG_LEVEL_INFO);
```

Comment:
Can we comment everything ! e.g. what is it for, why needed, how to use etc etc.